ceremonies/src/lib.rs

```diff
@@ -389,11 +390,93 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			cid: CommunityIdentifier,
 		) -> DispatchResultWithPostInfo {
-			let sender = ensure_signed(origin)?;
-			let (meetup_index, reward_count) =
-				Self::validate_one_meetup_and_issue_rewards(&sender, &cid)?;
+			let participant = &ensure_signed(origin)?;
+
+			if <encointer_scheduler::Pallet<T>>::current_phase() != CeremonyPhaseType::Registering {
+				return Err(<Error<T>>::WrongPhaseForClaimingRewards.into())
+			}
+
+			let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index() - 1; //safe; cindex comes from within
+			let meetup_index = Self::get_meetup_index((cid, cindex), participant)
+				.ok_or(<Error<T>>::ParticipantIsNotRegistered)?;
+
+			if <IssuedRewards<T>>::contains_key((cid, cindex), meetup_index) {
+				return Err(<Error<T>>::RewardsAlreadyIssued.into())
+			}
+			info!(
+				target: LOG,
+				"validating meetup {:?} for cid {:?} triggered by {:?}",
+				meetup_index,
+				&cid,
+				participant
+			);
 
-			Self::deposit_event(Event::RewardsIssued(cid, meetup_index, reward_count));
+			//gather all data
+			let meetup_participants = Self::get_meetup_participants((cid, cindex), meetup_index)
+				.ok_or(<Error<T>>::GetMeetupParticipantsError)?;
+			let (participant_votes, participant_attestations) =
+				Self::gather_meetup_validation_data(cid, cindex, meetup_participants.clone());
+
+			// initialize an array of local participant indices that are eligible for the reward
+			// indices will be deleted in the following based on various rules
+			let mut participants_eligible_for_rewards: Vec<usize> =
+				(0..meetup_participants.len()).collect();
+
+			let attestation_threshold_fn = |i: usize| max(i.saturating_sub(1), 1);
```

Comment:
I guess the right-hand side max should be 3. This is what we decided is the minimum meetup size for a successful meetup.

Reply:
@brenzi decided to leave it as is for the moment being